core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -580,23 +580,26 @@ public PlanOptimizers(
                                 new PruneTableScanColumns(metadata),
                                 new PushPredicateIntoTableScan(metadata, typeOperators, typeAnalyzer))));
 
+        // This excludes PushJoinIntoTableScan because it must run after ReorderJoins to not undo join reordering
+        // and after DetermineJoinDistributionType because DetermineJoinDistributionType requires
```

Comment:
In fact, both `DetermineJoinDistributionType` and `ReorderJoins` require stats which is the biggest factor here.

Also, connector still might provide stats after pushdown (stats are not lost), but most of them don't

Reply:
Will reword this to something like:

```
PushJoinIntoTableScan is excluded because it will prevent ReorderJoins and DetermineJoinDistributionType from running because the JoinNode will be consumed
```